### PR TITLE
AEA-1037: Restructure project.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

**Review by dan-jones-iw — APPROVED** [review]

File: coordinator/src/models/fhir/medication-request.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import * as common from "./common"
+import {createCodeableConcept} from "../../services/translation/response/fhir-base-types"
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/models/fhir/medication-request.ts:2]: Yep :+1: will move

File: coordinator/src/models/hl7-v3/application-acknowledgement.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+export enum AcknowledgementTypeCode {
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/models/hl7-v3/application-acknowledgement.ts:1]: There's a thing called application acknowledgement and this file is holding the definitions for that: https://data.developer.nhs.uk/dms/mim/4.2.00/Domains/Infrastructure/Tabular%20View/MCCI_HD020100UK12-NoEdit.htm

File: coordinator/src/models/fhir/message-header.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import * as extension from "./extension"
+
+export enum EventCodingCode {
+  PRESCRIPTION = "prescription-order",
+  PRESCRIPTION_RESPONSE = "prescription-order-response",
+  CANCELLATION = "prescription-order-update",
+  DISPENSE = "prescription-dispense"
+}
+
+export const EventCoding: Record<string, common.Coding> = Object.freeze({
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/models/fhir/message-header.ts:12]: That would be good, will try it out

File: coordinator/src/resources/MvpMedicationRequest.ts [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
-  CodeableConcept,
-  CodingExtension, Identifier,
-  IdentifierExtension,
   MedicationRequest,
-  MedicationRequestDispenseRequest, MedicationRequestGroupIdentifier
-} from "../models/fhir/fhir-resources"
+  MedicationRequestDispenseRequest,
+  MedicationRequestGroupIdentifier,
+  MedicationRequestStatus
+} from "../models/fhir/medication-request"
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/resources/MvpMedicationRequest.ts:7]: Might see if it's used first, but yeah

File: coordinator/src/models/fhir/message-header.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import * as common from "./common"
+import * as practitionerRole from "./practitioner-role"
+import * as extension from "./extension"
+
+export enum EventCodingCode {
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/models/fhir/message-header.ts:5]: Agh not sure. `MessageType` definitely has more meaning to us, but FHIR calls it `EventCoding`

[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/models/fhir/message-header.ts:5]: Looked it up in the IG and apparently it's called a Spine message event: https://simplifier.net/nhsdigitalspine/spine-message-events.  So maybe `SpineMessageEventCode`? I don't know if that's any better really.

**Pull request merged**